phpunit/functional/Glpi/Form/Migration/FormMigrationTest.php

```diff
@@ -2011,4 +2013,44 @@ public function testFormMigrationQuestionDropdownItemWithAdvancedOptions(): void
         $this->assertEquals($itilcategory->getId(), $question_type->getRootItemsId($question));
         $this->assertEquals(0, $question_type->getSubtreeDepth($question));
     }
+
+    public function testFormMigrationWithEntityThatDoesNotExist(): void
+    {
+        /**
+         * @var \DBmysql $DB
+         */
+        global $DB;
+
+        // Create a form
+        $this->assertTrue($DB->insert(
+            'glpi_plugin_formcreator_forms',
+            [
+                'name' => 'Test form migration with entity that does not exist',
+                "entities_id" => 999, // Non-existing entity ID
+            ]
+        ));
+
+        // Create a section for the form
+        $this->assertTrue($DB->insert(
+            'glpi_plugin_formcreator_sections',
+            [
+                'plugin_formcreator_forms_id' => $DB->insertId(),
+            ]
+        ));
+
+        // Process migration
+        $migration = new FormMigration($DB, FormAccessControlManager::getInstance());
+        $this->setPrivateProperty($migration, 'result', new PluginMigrationResult());
+        $this->assertTrue($this->callPrivateMethod($migration, 'processMigration'));
+
+        // Verify that the form has not been migrated
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('Unable to load a single `Glpi\Form\Form` item with the name `Test form migration with entity that does not exist` (none or many exist may exist).');
+        getItemByTypeName(Form::class, 'Test form migration with entity that does not exist');
```

Comment:
This should be more resilient to changes. Indeed, the exception message could slightly change int he future.

```suggestion
        $this->assertFalse((new Form())->getFromDBByCrit(['name' => 'Test form migration with entity that does not exist']));
```